beacon-chain/blockchain/process_block.go

```diff
@@ -107,6 +107,13 @@ func (s *Service) onBlock(ctx context.Context, signed block.SignedBeaconBlock, b
 		return err
 	}
 
+	// After saving the state, we add a proposer score boost to fork choice for the block root if applicable.
```

Comment:
I think doing this before `savePostStateInfo` is worth exploring. Although it diverts from the spec, it arguably more accurate from the protocol's standpoint since "saving the state" takes time